internal/translation/customroles/conversion.go

```diff
@@ -70,32 +70,38 @@ func toAtlasInheritedRoles(role *CustomRole) []admin.DatabaseInheritedRole {
 }
 
 func fromAtlas(role *admin.UserCustomDBRole) CustomRole {
-	inheritedRoles := make([]akov2.Role, 0, len(*role.InheritedRoles))
-
-	for _, atlasInheritedRole := range *role.InheritedRoles {
-		inheritedRoles = append(inheritedRoles, akov2.Role{
-			Name:     atlasInheritedRole.Role,
-			Database: atlasInheritedRole.Db,
-		})
+	var inheritedRoles []akov2.Role
+	if role.InheritedRoles != nil {
+		inheritedRoles = make([]akov2.Role, 0, len(*role.InheritedRoles))
+		for _, atlasInheritedRole := range *role.InheritedRoles {
+			inheritedRoles = append(inheritedRoles, akov2.Role{
+				Name:     atlasInheritedRole.Role,
+				Database: atlasInheritedRole.Db,
+			})
+		}
 	}
 
-	actions := make([]akov2.Action, 0, len(*role.Actions))
-
-	for _, atlasAction := range *role.Actions {
-		resources := make([]akov2.Resource, 0, len(*atlasAction.Resources))
+	var actions []akov2.Action
+	if role.Actions != nil {
+		actions = make([]akov2.Action, 0, len(*role.Actions))
+		for _, atlasAction := range *role.Actions {
+			var resources []akov2.Resource
+			if atlasAction.Resources != nil {
+				resources = make([]akov2.Resource, 0, len(*atlasAction.Resources))
+				for _, atlasResource := range *atlasAction.Resources {
+					resources = append(resources, akov2.Resource{
+						Cluster:    pointer.MakePtr(atlasResource.Cluster),
```

Comment:
Too much cyclomatic complexity here.

instead we could:
- Exit early returning the custom role without actions OR delegate the actions population to a function. 
- And/or have a function for each resource build.

Reply:
Agreed. But this complexity was there before. I just saved it from nil pointer deference that I got during testing. We can refactor this later if needed.